### PR TITLE
doc: add a section to troubleshooting about asset changes

### PR DESCRIPTION
**Pull request by eseidel**
Fixes https://github.com/shorebirdtech/docs/issues/162

### CODE REVIEW

File: docs/troubleshooting.md [hunk 1]
```diff
@@ -122,9 +122,63 @@ version and build numbers in order to target patches at specific releases. You
 can fix this problem by either setting `manageAppVersionAndBuildNumber` to false
 or removing the value from your export options .plist file.
 
+## I see a `The release artifact contains native changes` warning when running `shorebird release` (#native-changes)
```
Comment (bryanoltman) [inline docs/troubleshooting.md:125]: "native changes" -> "asset changes"

File: docs/troubleshooting.md [hunk 2]
```diff
@@ -122,9 +122,63 @@ version and build numbers in order to target patches at specific releases. You
...
+
+The `shorebird patch` command will print a warning if it detects changes to
+files in your compiled app that correspond to native code changes (`.dex` files
+on Android, files in the `Runner.app` directory on iOS). This does not always
+mean that your patch will not work, but because shorebird cannot be sure that
+the changes are safe, and because shorebird can't patch non-Dart code, it prints
+a warning.
```
Comment (bryanoltman) [inline docs/troubleshooting.md:132]: This seems like it's in the wrong section – the warning below is about asset changes, not native changes

Comment (eseidel) [reply docs/troubleshooting.md:132]: I think the warning says "native changes"

Comment (eseidel) [reply docs/troubleshooting.md:132]: Fixed.

**Comment by eseidel** [comment]
PTAL.

File: docs/troubleshooting.md [hunk 3]
```diff
@@ -122,9 +122,66 @@ version and build numbers in order to target patches at specific releases. You
 can fix this problem by either setting `manageAppVersionAndBuildNumber` to false
 or removing the value from your export options .plist file.
 
+## I see a `The release artifact contains asset changes` warning when running `shorebird release` (#asset-changes)
+
+The `shorebird patch` command will print a warning if it detects changes to
+files in your compiled app that correspond to asset changes (e.g. added or
+removed images, fonts, etc.). This does not always mean that your patch will not
+work, but because shorebird cannot be sure that the changes are safe.
```
Comment (bryanoltman) [inline docs/troubleshooting.md:130]:
```suggestion
work, but shorebird cannot be sure that the changes are safe.
```

**Review by bryanoltman — APPROVED** [review]

**Pull request merged**